### PR TITLE
feat(ui): add toggle button group component [LW-9269]

### PR DESCRIPTION
**Pull request by przemyslaw-wlodek**
# Checklist

- [x] [JIRA](https://input-output.atlassian.net/browse/LW-9269)
- [x] Proper tests implemented
- [x] Screenshots added.

---

## Proposed solution
Add a new ToggleButtonGroup component. Supporting keyboard navigation and other a11y-related features.

## Testing
Open storybook, test different options for user navigation and selection e.g. mouse clicks and keyboard focus.

## Screenshots
image
image

### CODE REVIEW

File: packages/ui/src/assets/icons/key.component.svg
Comment (mchappell) [inline packages/ui/src/assets/icons/key.component.svg:1]: Please put this in a separate nested folder to denote that it's not actually used in the app

File: packages/ui/src/assets/icons/home.component.svg
Comment (mchappell) [inline packages/ui/src/assets/icons/home.component.svg:1]: Please put this in a separate nested folder to denote that it's not actually used in the app

File: packages/ui/src/assets/icons/globe-alt.component.svg
Comment (mchappell) [inline packages/ui/src/assets/icons/globe-alt.component.svg:1]: Please put this in a separate nested folder to denote that it's not actually used in the app

File: packages/ui/src/design-system/toggle-button-group/toggle-button-group-item.component.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import * as ToggleGroup from '@radix-ui/react-toggle-group';
+
+import * as cx from './toggle-button-group-item.css';
+
+import type { ToggleGroupItemProps } from '@radix-ui/react-toggle-group';
+
+export type ToggleButtonGroupItemProps = Pick<
+  ToggleGroupItemProps,
+  'children' | 'disabled' | 'id' | 'value'
```
Comment (mchappell) [inline packages/ui/src/design-system/toggle-button-group/toggle-button-group-item.component.tsx:11]: can we add a `dataTestId` attribute here to make e2e targeting easier

File: packages/ui/src/design-system/toggle-button-group/toggle-button-group-root.component.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import * as ToggleGroup from '@radix-ui/react-toggle-group';
+import classNames from 'classnames';
+
+import * as cx from './toggle-button-group-root.css';
+
+import type { ToggleGroupSingleProps } from '@radix-ui/react-toggle-group';
+
+type ToggleGroupSingleOptionalProps = Pick<
+  ToggleGroupSingleProps,
+  'loop' | 'rovingFocus'
```
Comment (mchappell) [inline packages/ui/src/design-system/toggle-button-group/toggle-button-group-root.component.tsx:12]: can we add a `dataTestId` attribute here to make e2e targeting easier

**Review by mchappell — CHANGES_REQUESTED** [review]

File: packages/ui/src/design-system/toggle-button-group/toggle-button-group-item.component.tsx (hunk 1 above)
Comment (przemyslaw-wlodek) [reply packages/ui/src/design-system/toggle-button-group/toggle-button-group-item.component.tsx:11]: All data attributes can be placed without explicit mention in props :) I also include the example in storybook e.g. `data-testid="tab1-label-only"`

File: packages/ui/src/design-system/toggle-button-group/toggle-button-group-root.component.tsx (hunk 2 above)
Comment (przemyslaw-wlodek) [reply packages/ui/src/design-system/toggle-button-group/toggle-button-group-root.component.tsx:12]: same as above, thanks :)

**Review by mchappell — APPROVED** [review]

**Comment by mchappell** [comment]
Nice work @przemyslaw-wlodek 💪

**Pull request merged**